test/e2e/funcs/feature.go

```diff
@@ -286,6 +288,11 @@ func ResourcesHaveFieldValueWithin(d time.Duration, dir, pattern, path string, w
 			return ctx
 		}
 
+		if count.Load() == 0 {
```

Comment:
Didnt dig into it, but shouldn't the `wait.For(..` return an err if no resources matched the pattern in the given time?

Reply:
Yes, but not if you asked it to wait for no resources at all in the first place 😉